### PR TITLE
User Directory: Elastic Search Edition.

### PR DESCRIPTION
**Pull request by androolloyd**
##### Description
- indexes the UserDirectory view into the elastic search store,
- uses django haystack
- new settings required for connection through haystack 
- new staff protected route /user_directory
- staff can export to csv

### CODE REVIEW

File: app/assets/v2/js/users.js [hunk 1]
```diff
@@ -347,73 +410,90 @@ Vue.component('user-directory', {
...
+    this.fetchMappings();
+    this.fetchUsers();
+    this.$watch('params', function(newVal, oldVal) {
+      this.searchUsers();
+    }, {
+      deep: true
+    });
+  },
+  created() {
+    this.setHost('https://elastic.androolloyd.com'); // TODO: set to proper env variable
```
Comment (owocki) [inline app/assets/v2/js/users.js:477]: hardcoded

File: app/dashboard/router.py [hunk 2]
```diff
@@ -285,6 +295,112 @@ def get_queryset(self):
...
+        order_by = self.request.query_params.get('order_by', '-follower_count')
+        bounties_completed = self.request.query_params.get('bounties_completed', '').strip().split(',')
+        leaderboard_rank = self.request.query_params.get('leaderboard_rank', '').strip().split(',')
+        rating = int(self.request.query_params.get('rating', '0'))
+        organisation = self.request.query_params.get('organisation', '')
+        tribe = self.request.query_params.get('tribe', '')
+        hackathon_id = self.request.query_params.get('hackathon', '')
+        user_filter = self.request.query_params.get('user_filter', '')
+
+        # user_id = self.request.query_params.get('user', None)
```
Comment (danlipert) [inline app/dashboard/router.py:319]: some leftover code here

File: app/dashboard/router.py [hunk 3]
```diff
@@ -285,6 +295,112 @@ def get_queryset(self):
...
+        current_user = self.request.user if hasattr(self.request, 'user') and self.request.user.is_authenticated else None
+
+        # if not current_user:
+        #     return redirect('/login/github?next=' + request.get_full_path())
+        current_profile = Profile.objects.get(user=current_user)
+
+        # if not settings.DEBUG:
+        #     network = 'mainnet'
+        # else:
+        network = 'rinkeby'
```
Comment (danlipert) [inline app/dashboard/router.py:332]: network hardcoded

File: app/dashboard/router.py [hunk 4]
```diff
@@ -643,3 +759,4 @@ class Meta:
 
 router.register(r'bounty', BountyViewSet)
 router.register(r'projects_fetch', HackathonProjectsViewSet)
+# router.register(r'users_fetch', UserDirectoryViewSet)
```
Comment (danlipert) [inline app/dashboard/router.py:762]: Viewset commented out

File: docker-compose.yml [hunk 5]
```diff
@@ -81,18 +90,18 @@ services:
...
-      - "8545:8545"
-    command: '-a 1 --seed 2 --host 0.0.0.0 --mnemonic "${TEST_MNEMONIC}"'
-    deploy:
-      replicas: 1
-      resources:
-        limits:
-          memory: 1024M
-        reservations:
-          memory: 128M
+#  testrpc:
```
Comment (danlipert) [inline docker-compose.yml:93]: Let's restore this in this PR, but we should thing about removing it from the dockerfile for sure

File: elasticsearch.yml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+network.host: 0.0.0.0
```
Comment (danlipert) [inline elasticsearch.yml:1]: Is this just for the docker container in local?

**Review by danlipert — CHANGES_REQUESTED** [review]
Looks like its almost there! excited to get this out

File: elasticsearch.yml (hunk 6 above)
Comment (androolloyd) [reply elasticsearch.yml:1]: yes its

File: app/dashboard/router.py (hunk 4 above)
Comment (androolloyd) [reply app/dashboard/router.py:762]: we're not using the users fetch here.

**Review by thelostone-mc — APPROVED** [review]

**Pull request merged**